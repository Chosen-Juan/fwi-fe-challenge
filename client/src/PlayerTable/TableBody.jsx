import React from 'react';
import PropTypes from 'prop-types';
import Flags from 'react-world-flags';

import Avatar from '../Avatar';
import { COUNTRIES } from '../constants';

const TableBody = ({ players }) => {
  return (
    <tbody className="table--body">
      {players.map(({ id, name, country, winnings, imageUrl }) => (
        <tr key={id} role="row" className="table__row">
          <td>
            <Avatar src={imageUrl} />
          </td>
          <td className="centered-text">
            {name}
          </td>
          <td className="centered-text">
            {winnings.toLocaleString(undefined, {
              style: 'currency',
              currency: 'USD',
            })}
          </td>
          <td>
            <div className="country">
              <Avatar>
                <Flags code={country} alt="" />
              </Avatar>
              {country}
            </div>
          </td>
        </tr>
      ))}
    </tbody>
  );
};

TableBody.propTypes = {
  players: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      country: PropTypes.oneOf(Object.keys(COUNTRIES)),
      winnings: PropTypes.number.isRequired,
      imageUrl: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default TableBody;
