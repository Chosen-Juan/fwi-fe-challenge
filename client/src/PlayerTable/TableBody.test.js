import React from 'react';
import { shallow } from 'enzyme';

import TableBody from './TableBody';

describe('TableBody', () => {
  const defaultProps = {
    players: [{
      id: '1one',
      name: 'Jon Doe',
      country: 'US',
      winnings: 24999,
      imageUrl: ''
    }]
  };
  const tableBody = shallow(<TableBody {...defaultProps} />);

  it('should match snapshot', () => {
    expect(tableBody).toMatchSnapshot();
  });

});
