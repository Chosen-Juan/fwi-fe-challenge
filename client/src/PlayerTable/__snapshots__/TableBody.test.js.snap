// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody should match snapshot 1`] = `
<tbody
  className="table--body"
>
  <Connect(Player)
    key="1one"
    player={
      Object {
        "country": "US",
        "id": "1one",
        "imageUrl": "",
        "name": "Jon Doe",
        "winnings": 24999,
      }
    }
  />
</tbody>
`;
